packages/venia-ui/lib/RootComponents/Search/search.js

```diff
@@ -1,181 +1,154 @@
-                        {({ loading, error, data }) => {
-                            if (error) return null;
-                            if (loading) return 'Loading...';
-                            return data.category.name;
-                        }}
-                    </Query>
-                </small>
-                <Icon
-                    src={CloseIcon}
-                    attrs={{
-                        width: '13px',
-                        height: '13px'
-                    }}
-                />
-            </button>
-        </div>
+                        Filter
+                    </button>
+                </div>
+            ) : null;
+
+            const maybeFilterModal = filters ? (
+                <FilterModal filters={filters} />
+            ) : null;
+
+            return (
+                <article className={classes.root}>
+                    <div className={classes.categoryTop}>
+                        <div className={classes.totalPages}>
+                            {`${total_count} items`}
+                        </div>
+                        {maybeCategoryFilters}
+                        {maybeFilterButtons}
+                    </div>
+                    {maybeFilterModal}
+                    <section className={classes.gallery}>
+                        <Gallery data={items} />
+                    </section>
+                </article>
+            );
+        },
+        [categoryId, classes, executeSearch, history, location, openDrawer]
     );
 
-    handleClearCategoryFilter = () => {
-        const inputText = getQueryParameterValue({
-            location: this.props.location,
-            queryParameter: 'query'
-        });
+    // derive initial state from query params
+    // never re-run this effect, even if deps change
+    /* eslint-disable react-hooks/exhaustive-deps */
+    useEffect(() => {
+        // clear filters if there are no filter params
+        if (isObjectEmpty(getFilterParams())) {
+            filterClear();
+        }
 
-        if (inputText) {
-            this.props.executeSearch(inputText, this.props.history);
+        // ensure search is open to begin with
+        if (toggleSearch && !searchOpen && inputText) {
+            toggleSearch();
         }
-    };
-
-    render() {
-        const { classes, location, openDrawer } = this.props;
-        const { getCategoryName } = this;
-
-        const inputText = getQueryParameterValue({
-            location,
-            queryParameter: 'query'
-        });
-        const categoryId = getQueryParameterValue({
-            location,
-            queryParameter: 'category'
-        });
-
-        if (!inputText) {
-            return <Redirect to="/" />;
+    }, []);
+    /* eslint-enable react-hooks/exhaustive-deps */
+
+    // clear filters whenever the `query` search param changes
+    // but don't clear them on mount
+    useEffect(() => {
+        if (shouldClear.current) {
+            filterClear();
```

Comment:
We also shouldn't need to clear if it's already clear right? ie I start typing in a new search input, filter gets cleared after one character and re-cleared every subsequent character because of this effect.

Reply:
No, this effect doesn't fire when the text field value changes; it fires when the search param value changes. [Compare to the original.](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/RootComponents/Search/search.js#L42-L71)

Reply:
`inputText` and `queryText` confuses me :D